src/graphql/schema/node/createNodeFieldEntry.ts

```diff
@@ -46,7 +46,7 @@ export default function createNodeFieldEntry (buildToken: BuildToken): [string,
       if (!primaryKey || !primaryKey.read)
         throw new Error(`Invalid id, no readable primary key on collection named '${collection.name}'.`)
 
-      const value = await primaryKey.read(context, keyValue)
+      const value = await primaryKey.read(context, keyValue, resolveInfo, getGqlOutputType(buildToken, collection.type).gqlType)
```

Comment:
`getNullableType` from `graphql` is really great for getting the nullable GraphQL type, and I believe `src/interface` has a `getNonNullType` function because nullability is reversed in `src/interface` 😊

Here we could build an index keyed by the collection’s name of type `Map<string, GraphQLObjectType>`. I think I’d feel better if we did that. I don’t really want to make any guarantees that `getGqlOutputType` and similar functions are fast enough for runtime 😊